l2geth/core/state_transition.go

```diff
@@ -173,6 +184,15 @@ func (st *StateTransition) buyGas() error {
 
 	st.initialGas = st.msg.Gas()
 	st.state.SubBalance(st.msg.From(), mgval)
+
+	if rcfg.UsingOVM {
```

Comment:
I think it would be cleaner to add this to mgval above instead of doing this here

Reply:
I.e., if UsingOVM { mgval = mgval + st.l1Fee }

Reply:
I debated this with @ben-chain a bit, it was one of the approaches that we considered